src/NodeServices/GraphicsDataInterfaces.cs

```diff
@@ -211,6 +211,49 @@ public interface IGraphicItem
         void Tessellate(IRenderPackage package, TessellationParameters parameters);
     }
 
+    /// <summary>
+    /// An interface that defines items which have a transform property which is a 4x4 matrix
+    /// </summary>
+    public interface ITransformable
+    {
+        /// <summary>
+        /// A 4x4 matrix which is used to transform all geometry in the render packaage
+        /// </summary>
+        double[] Transform  { get; } 
+        
+        /// <summary>
+        /// Sets the transform as a series of doubles that will be applied to all geometry in the renderPackage
+        /// </summary>
+        /// <param name="m11"></param>
+        /// <param name="m12"></param>
+        /// <param name="m13"></param>
+        /// <param name="m14"></param>
+        /// <param name="m21"></param>
+        /// <param name="m22"></param>
+        /// <param name="m23"></param>
+        /// <param name="m24"></param>
+        /// <param name="m31"></param>
+        /// <param name="m32"></param>
+        /// <param name="m33"></param>
+        /// <param name="m34"></param>
+        /// <param name="m41"></param>
+        /// <param name="m42"></param>
+        /// <param name="m43"></param>
+        /// <param name="m44"></param>
+        void SetTransform(double m11, double m12, double m13, double m14,
+           double m21, double m22, double m23, double m24,
+           double m31, double m32, double m33, double m34,
+           double m41, double m42, double m43, double m44);
+
+        /// <summary>
+        /// Sets the transform as a double array, this transform will be applied to all geometry in the renderPackage
+        /// </summary>
+        /// <param name="matrix"></param>
+        void SetTransform(double[] matrix);
```

Comment:
Can you specify how the array should be laid out? Like "Array should be specified in row order X1, Y1, Z1, X2, Y2,...". Not sure if our coordinate systems are row-ordered or column-ordered. This will reduce the amount of guessing by the user.